app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/SelectionInteractionViewModel.kt

```diff
@@ -118,30 +118,31 @@ class SelectionInteractionViewModel(
 
   /** Catalogs an item being clicked by the user and returns whether the item should be considered selected. */
   fun updateSelection(itemIndex: Int, isCurrentlySelected: Boolean): Boolean {
-    if (isCurrentlySelected) {
-      selectedItems -= itemIndex
-      updateIsAnswerAvailable()
-      return false
-    } else {
-      if (areCheckboxesBound()) {
-        if (selectedItems.size < maxAllowableSelectionCount) {
-          // TODO(#3624): Add warning to user when they exceed the number of allowable selections or are under the minimum
-          //  number required.
-          selectedItems += itemIndex
-          updateIsAnswerAvailable()
-          return true
-        }
-      } else {
+    when {
+      isCurrentlySelected -> {
+        selectedItems -= itemIndex
+        updateIsAnswerAvailable()
+        return false
+      }
+      !areCheckboxesBound() -> {
         // Disable all items to simulate a radio button group.
         choiceItems.forEach { item -> item.isAnswerSelected.set(false) }
         selectedItems.clear()
         selectedItems += itemIndex
         updateIsAnswerAvailable()
         return true
       }
+      selectedItems.size < maxAllowableSelectionCount -> {
+        // TODO(#3624): Add warning to user when they exceed the number of allowable selections or are under the minimum
+        //  number required.
+        selectedItems += itemIndex
+        updateIsAnswerAvailable()
+        return true
```

Comment:
The 'return' can be 'lifted' out of the when.

Reply:
Thought to keep to make things more readable.

Lifted out now.